Tests/MapboxSearchIntegrationTests/MockServerIntegrationTestCase.swift

```diff
@@ -2,10 +2,11 @@ import CoreLocation
 @testable import MapboxSearch
 import XCTest
 
-class MockServerIntegrationTestCase: XCTestCase {
-    let server = MockWebServer()
+class MockServerIntegrationTestCase<Mock: MockResponse>: XCTestCase {
+    typealias Mock = Mock
```

Comment:
nit: looks like we can remove explicit typealias declaration

Reply:
That would cause the Test subclasses to need either:
1. declare a name in the generic, such as `<Mock: MockResponse>` (more noise, but explicit which is nice) or 
2. change `Mock` usages to the default name `T`, such as `T.coreApiType.toSDKType()` (more obtuse IMHO)

The typealias is inherited and I think it's the simplest approach (although you do have to visit the parent class definition to find the typealias.) Let me know what you think.